src/tls_wrap.h

```diff
@@ -132,6 +132,7 @@ class TLSWrap : public AsyncWrap,
 
   AsyncWrap* GetAsyncWrap() override;
   bool IsIPCPipe() override;
+  void UpdateWriteQueueSize(int write_queue_size = 0);
```

Comment:
It seems like `0` might be a valid length for the write queue size (now or at least conceptually/in the future?), so maybe this could be `-1` for indicating a fallback/default?

Reply:
The idea here is that it's harmful for it to be set to 0 if `BIO_pending(enc_out_)` is not actually 0 — it can then lead to breaking the logic in `net.js`.

Reply:
@apapirovski As far as I understand it, `write_queue_size = 0` is used to say “use the value provided by `BIO_pending`”? And `BIO_pending` can return 0, right? In that case I’d say using `-1` would be a better value to communicate “please fall back to `BIO_pending`”

Reply:
Right but we want to disallow someone forcing through `UpdateWriteQueueSize(0)` because if they do that and BIO_pending is not actually 0 then they potentially just broke stuff.

Reply:
> Right but we want to disallow someone forcing through `UpdateWriteQueueSize(0)` because if they do that and BIO_pending is not actually 0 then they potentially just broke stuff.

Okay, that makes a bit more sense, but could you explain why?

Reply:
`net.js` depends on the writeQueueSize being > 0 if the write hasn't been flushed (`BIO_pending` returns > 0). I haven't followed the trail all the way but in my testing any time that wasn't the case, it broke everything.

It's also the reason we have to set `writeQueueSize` to 1 for empty writes. It's possible all that logic could be further improved but it would require far more significant changes than this.